### PR TITLE
🧹 Centralize date and duration formatting in `src/lib/utils.ts`

### PR DESCRIPTION
**Pull request by rube-de**
🎯 **What:** Moved `formatDate` and `formatDuration` logic from `src/components/library/saved-episode-card.tsx` to `src/lib/utils.ts` and updated usages.
💡 **Why:** Reduces code duplication and ensures consistent formatting across the application.
✅ **Verification:** Added new unit tests in `src/lib/__tests__/utils.test.ts` and verified existing tests pass.
✨ **Result:** Cleaner component code and reusable utilities.

---
*PR created automatically by Jules for task [13294378590621733596](https://jules.google.com/task/13294378590621733596) started by @funny-otter*

## Summary by CodeRabbit

* **Refactor**
  * Consolidated date and duration formatting logic into reusable utilities for improved maintainability and consistency across the application.

* **Tests**
  * Added comprehensive test coverage for date and duration formatting utilities, including edge case handling.

### CODE REVIEW

File: src/lib/__tests__/utils.test.ts [hunk 1]
```diff
@@ -58,3 +58,47 @@ describe("stripHtml", () => {
...
+    const result = formatDate(d);
+    expect(result).toMatch(/Oct/);
+    expect(result).toMatch(/2023/);
+  });
+
+  it("handles null/undefined", () => {
+    expect(formatDate(null)).toBe("");
+    expect(formatDate(undefined)).toBe("");
+  });
+});
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/lib/__tests__/utils.test.ts]: Fixed: Added day assertions to existing tests for specificity. Added edge case tests for `formatDate(0)` (Unix epoch) and `formatDate("not-a-date")` (invalid input).

File: src/lib/utils.ts [hunk 2]
```diff
@@ -8,3 +8,22 @@ export function cn(...inputs: ClassValue[]) {
...
+
+export function formatDuration(seconds: number | null | undefined): string {
+  if (!seconds || seconds <= 0) return "";
+  const hours = Math.floor(seconds / 3600);
+  const minutes = Math.floor((seconds % 3600) / 60);
+  if (hours > 0) {
+    return `${hours}h ${minutes}m`;
+  }
+  return `${minutes}m`;
+}
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/lib/utils.ts]: Flagged for human review — see PR check summary below.

File: src/lib/utils.ts [hunk 3]
```diff
@@ -8,3 +8,22 @@ export function cn(...inputs: ClassValue[]) {
...
+
+export function formatDate(date: Date | string | number | null | undefined): string {
```
[the remark this answers is not in the document]
Comment (funny-otter) [reply src/lib/utils.ts]: Fixed: Added JSDoc to `formatDate` clarifying the `number` parameter expects milliseconds. Added a new `formatDateFromUnix(timestamp)` helper that handles the seconds-to-ms conversion, which future callers migrating from PodcastIndex timestamps should use.
